src/utils/git.rs

```diff
@@ -0,0 +1,35 @@
+use std::process::Command;
+
+pub fn retrieve_git_version() -> Option<(usize, usize)> {
+    if let Ok(git_path) = grep_cli::resolve_binary("git") {
+        let cmd = Command::new(git_path).arg("--version").output().ok()?;
+        parse_git_version(&cmd.stdout)
+    } else {
+        None
+    }
+}
```

Comment:
FWIW, this takes 6.5ms on an M2 Mac (hyperfine's estimate based on 1000 calls in a loop), so slightly slower than the 4ms you found for `less`. (But, if neither file uses process substitution then it's not done at all)